_posts/2023/2023-06-06-getting-started-with-vector-dbs-in-python.md

```diff
@@ -221,8 +221,163 @@ USERNAME=admin PASSWORD=admin ENDPOINT=https://localhost:9200 poetry run src/ope
 {'total': {'value': 1, 'relation': 'eq'}, 'max_score': 0.97087383, 'hits': [{'_index': 'my-index', '_id': 'vec1', '_score': 0.97087383, '_source': {'index': {'_index': 'my-index', '_id': 'vec2'}, 'vector': [0.2, 0.3, 0.4], 'metadata': {'genre': 'action'}}}]}
 {% endhighlight %}
 
+### Vespa
+
+[Vespa](https://vespa.ai/) is a fully featured search engine and vector database. It supports vector search (ANN), lexical search, and search in structured data, all in the same query. Vespa is Apache 2.0 licensed, and can be run in a variety of ways, including Docker and as a managed [cloud service](https://cloud.vespa.ai/).
+
+Let's use their Docker container for this example.
+
+{% highlight bash %}
+docker info | grep "Total Memory" # make sure it's at least 4Gb
+docker run --detach --name vespa --hostname vespa-container \
+  --publish 8080:8080 --publish 19071:19071 \
+  vespaengine/vespa
+{% endhighlight %}
+
+This container listsens on port `8080` for search and and ingestion APIs and on `19071` for configuration APIs.
+
+Vespa encapsulates the concept of a schema/index in an application that needs to be defined and deployed, so it is not as straightforward as the previous example.
+
+To create a new application with a sample vector schema we need to create a `settings.xml` file with the overall application properties and a `schema.md` file with the definition of our schema. For this example, let's create the following directory structure
```

Comment:
Since it's a full sentence finish with a period. (structure.)

Reply:
Updated.